File: python/minimumSizeSubArraySum.py
def minimumSizeSubArraySum(s, nums):
    sum = 0;
    wIndex = 0;
    index = 0;
    subArrLen = len(nums) + 1;

    while index < len(nums):
        if sum > s:
            sum = sum - nums[wIndex]
            wIndex = wIndex + 1
        else:
            if (sum == s):
                current = index - wIndex
                if (subArrLen > current):
                    subArrLen = current
                
            sum = sum + nums[index]
            index = index + 1
    
    if (subArrLen > len(nums)): return 0
    else: return subArrLen


print(minimumSizeSubArraySum(7, [3,4,2,1,3,2]))

File: js/minimum_size_subarray_sum.js
function minimum_size_subarray_sum(s, list) {
  let wIdx = 0;
  let subArrLen = Infinity;
  for (let i = 1; i < list.length; i++) {
    if (list[wIdx] + list[i] === s) {
      if (subArrLen > i - wIdx + 1) subArrLen = i - wIdx + 1;
    }

    if (list[wIdx] + list[i] > s) {
      wIdx++;
    }
  }
  return subArrLen === Infinity ? -1 : subArrLen;
}

console.log(minimum_size_subarray_sum(7, [3, 4, 2, 1, 3, 2]));
